lib/streamlit/runtime/state/query_params_proxy.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Iterator, List, Union
+
+from streamlit.runtime.state.query_params import (
+    QueryParams,
+    _missing_key_error_message_query_params,
+)
+from streamlit.runtime.state.session_state_proxy import get_session_state
+from streamlit.type_util import Key
+
+
+def get_query_params() -> QueryParams:
+    from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+    ctx = get_script_run_ctx()
+
+    if ctx is None:
+        return QueryParams()
+    return get_session_state()._state._query_params
+
+
+class QueryParamsProxy:
+    """A stateless singleton that proxies `st.query_params` interactions
+    to the current script thread's QueryParams instance.
+    """
+
+    def __iter__(self) -> Iterator[Any]:
+        return iter(get_query_params())
+
+    def __len__(self) -> int:
+        return len(get_query_params())
+
+    def __getitem__(self, key: str) -> str:
+        return get_query_params()[key]
+
+    def __getattr__(self, key: str) -> str:
+        try:
+            return get_query_params()[key]
+        except KeyError:
+            raise AttributeError(_missing_key_error_message_query_params(key))
```

Comment:
We should just define this method using `getattr` so that we can just use the definition of `__getattr__` already defined in the `QueryParams` class. Otherwise, there's no point in defining it in the inner class since its' not being used at the moment anyway